include/arch/arc/v2/irq.h

```diff
@@ -84,7 +101,9 @@ extern void z_irq_spurious(const void *unused);
 	"irq priority cannot be set to 0 when CONFIG_ARC_FIRQ_STACK" \
 	"is not configured or CONFIG_ARC_FIRQ_STACK " \
 	"and CONFIG_ARC_STACK_CHECKING are configured together"); \
-	z_irq_priority_set(irq_p, priority_p, flags_p); \
```

Comment:
Some comments why priority cannot be set here and is moved would be good (I don't really understand why it cannot be left here). Also, above comments might not be applicable at this location, when initialization is moved (not sure I understand the comment, or why it is a comment and not an assert).

Reply:
Updated the comment for ARCH_IRQ_CONNECT, thanks!